Added counter of processed files and output of approximate remaining time.

andreili opened the pull request:
The code was rewritten from the old version of the converter.
Example output:
```
D:\Projects\waifu2x-converter-cpp\output\Release>waifu2x-converter-cpp.exe --mode noise_scale -q -i .\to_conv\ -o .\out\
CUDA: GeForce GTX 960
We're going to be operating in a directory. dir:D:\Projects\waifu2x-converter-cpp\output\Release\.\to_conv\
[1/4] 024f1c79df86.jpg (elapsed: 4s file: 1.38006s avg: 1.44535s)
[2/4] 03f662LvrLo.jpg (elapsed: 1s file: 0.269408s avg: 0.858336s)
[3/4] 0yLq0KXk8cY.jpg (elapsed: 0s file: 0.269567s avg: 0.663074s)
[4/4] 12063214_463445740528642_1095354792_n.jpg (elapsed: 0s file: 0.68575s avg: 0.669358s)
process successfully done! (all:2.67904[sec], 4 [files processed], 0 [files errored], 741.043[GFLOPS], filter:2.22665[sec], 891.603[GFLOPS])
```

DeadSix27 commented on src/main.cpp:
```diff
@@ -361,37 +362,60 @@ int main(int argc, char** argv) {
+			++numFilesProcessed;
+			double time_file_start = getsec();
 
+			std::cout << "[" << numFilesProcessed << "/" << files_count << "] " << fn.filename();
```
A linebreak `<< std::endl;` is missing here.

andreili replied:
It is possible and so, but initially in my version there were no new lines - everything was displayed in 1 line:
```
//std::cout << std::flush << "\r" << numFilesProcessed << "/" << files_count << ": " << input_file << " (elapsed: ";
```

DeadSix27 replied:
Well it produces this currently:
![image](https://user-images.githubusercontent.com/1492505/33574871-0c5f2706-d93b-11e7-8f31-36f28cc25dc2.png)
Doesn't look very nice

andreili replied:
yes, in that case it will not look very nice, you're right ...

andreili commented:
Delete latest commit, i'm a stupid :(

DeadSix27 commented:
@andreili There's no delete, just revert it or add a new commit with the fix :P

andreili commented:
@DeadSix27 I'm not yet very experienced in the git ;) But now I'm learning from mistakes...

DeadSix27 commented:
@andreili I learned that there is no delete the hard way too.. often kinda annoying.
But just remove the duplicate end1 and commit it. That's the right approach here

andreili commented:
It's finished ;)

DeadSix27 commented:
@andreili the linebreak is still missing.. lol

andreili commented:
dammit, second...

andreili commented:
Or this option, more versatile - adds line wrapping only if the output is detailed.
```
			std::cout << "[" << numFilesProcessed << "/" << files_count << "] " << fn.filename() << (verbose ? std::endl : "");
```

DeadSix27 commented:
@andreili In that case something like this: `(verbose ? std::endl : ": ")` would be better otherwise it would look ugly again

DeadSix27 commented:
@andreili Oh and if you wrap it all in one line when verbosity is low, maybe also remove the brackets.

E.g currently it's:
```
[2/2] "b7dsa.png"
 (elapsed: 0s file: 0.255851s avg: 0.282158s)
```
Your change would make it:
```
[2/2] "b7dsa.png"(elapsed: 0s file: 0.255851s avg: 0.282158s)
```
But this would look the best imo:
```
[2/2] "b7dsa.png": elapsed: 0s, file: 0.255851s, avg: 0.282158s
```

andreili commented:
You're right. But there is no time to reign, only if tomorrow = /

DeadSix27 commented:
@andreili Well, add: `(verbose ? std::endl : "")` and i'll merge it then you can change the rest tomorrow.

andreili commented:
Hmmm...
Added the output "ok" (if not verbose) and removed the brackets in the output.
![image](https://user-images.githubusercontent.com/1331777/33576531-2c0aac86-d951-11e7-9eb7-890e5331a468.png)

DeadSix27 commented:
@andreili 
```
/home/*/aur/waifu2x-converter-cpp/src/main.cpp: In function ‘int main(int, char**)’:
/home/*/aur/waifu2x-converter-cpp/src/main.cpp:390:115: error: overloaded function with no contextual type information
    std::cout << "[" << numFilesProcessed << "/" << files_count << "] " << fn.filename() << (verbose ? std::endl : " Ok. ");
                                                                                                                   ^~~~~~~
make[2]: *** [CMakeFiles/waifu2x-converter-cpp.dir/build.make:63: CMakeFiles/waifu2x-converter-cpp.dir/src/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:179: CMakeFiles/waifu2x-converter-cpp.dir/all] Error 2

```

andreili commented:
C++ does not support unary operations on output via cout :(

DeadSix27 commented:
@andreili It does not, the issue is that std::end1 is not a string, it adds a line break and then flushes the console, so using ` << (verbose? "\n" : " OK: ")` would work.

andreili commented:
It is more correct to use std::endl with std::cout.
https://stackoverflow.com/questions/213907/c-stdendl-vs-n

DeadSix27 commented:
True but still:
"Both serve the same purpose, putting a new line. The only difference is that endl causes flushing of the output buffer everytime it is called where as ‘\n’ does not."

Only flush is the difference, up to you if you prefer having it all in 1 but sacrifice the flush or not.

DeadSix27 commented:
Thanks :)

iame6162013 commented:
@DeadSix27 @andreili 

> There's no delete, just revert it or add a new commit with the fix :P

Depending on what you want, you can delete a commit via history rewriting.
http://christoph.ruegg.name/blog/git-howto-revert-a-commit-already-pushed-to-a-remote-reposit.html

DeadSix27 commented:
@iame6162013 Well yeah, same approach as here. https://help.github.com/articles/removing-sensitive-data-from-a-repository/
But it's so much work and so ugly, it's better to just revert or make a new commit with a fix
